contracts/modules/royalty/policies/IpPool.sol

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
+
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { ReentrancyGuard } from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+import { ERC20Snapshot, ERC20 } from "@openzeppelin/contracts-v4/token/ERC20/extensions/ERC20Snapshot.sol";
+import { SafeERC20 } from "@openzeppelin/contracts-v4/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+
+import { IRoyaltyPolicyLAP } from "../../../interfaces/modules/royalty/policies/IRoyaltyPolicyLAP.sol";
+import { IIpPool } from "../../../interfaces/modules/royalty/policies/IIpPool.sol";
+import { ArrayUtils } from "../../../lib/ArrayUtils.sol";
+import { Errors } from "../../../lib/Errors.sol";
+
+/// @title Ip Pool
+/// @notice Defines the logic for claiming royalty tokens and revenue tokens for a given IP
+contract IpPool is IIpPool, ERC20Snapshot, ReentrancyGuard {
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using SafeERC20 for IERC20;
+
+    /// @notice LAP royalty policy address
```

Comment:
Is it upgradable?  You may want to group the variables

Reply:
The basic implementation of IP Pool is not upgradable. But we might want to make it upgradable, cc @Ramarti, which requires Beacon Proxy cc @Spablob

Reply:
Yes upgradeability will be in a separate pr after this one